framework/SupervisedLearning/ARMA.py

```diff
@@ -360,7 +360,7 @@ def __trainLocal__(self,featureVals,targetVals):
         self.fourierResults[target] = self._trainFourier(self.pivotParameterValues,
                                                          self.fourierParams[target],
                                                          timeSeriesData,
-                                                         masks=[maskPeakRes],  # In future, a consolidated masking system for multiple signal processors can be implemented.
+                                                         masks=np.logical_and(maskPeakRes,self.notZeroFilterMask),  # In future, a consolidated masking system for multiple signal processors can be implemented.
```

Comment:
What you have here looks like it will work well. Do you think it might work better as a list of masks, then logically combine them within individual methods? That way we could carry around a list of masks in `__trainLocal__` and append to it as needed. Thoughts?

Reply:
We could even write a method in mathUtils or just in ARMA for combining the masks, so you don't have to rewrite it each time.